sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/AnalyticsTransmissionTarget.java

```diff
@@ -24,9 +40,11 @@
      * Create a new instance.
      *
      * @param transmissionTargetToken The token for this transmission target.
+     * @param parentTarget            Parent transmission target.
      */
-    AnalyticsTransmissionTarget(@NonNull String transmissionTargetToken) {
+    AnalyticsTransmissionTarget(@NonNull String transmissionTargetToken, final AnalyticsTransmissionTarget parentTarget) {
         mTransmissionTargetToken = transmissionTargetToken;
+        mParentTarget = parentTarget;
```

Comment:
Shouldn't we initialize the state of this transmission target with the state of its parents to align both? Then later we could trust the immediate parent state without checking other parents.

Reply:
We can't do that in the main thread in Android so it would actually be more complex as you would have to post a command and do the get and set in background, and in the meantime you can add children before the background task executes so you would probably do too many storage operations if that happens.

Reply:
I see, we don't have async public APIs on iOS yet so it would work there.